compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLocation.java

```diff
@@ -42,6 +42,13 @@ public BLangDiagnosticLocation(String filePath, int startLine, int endLine, int
         this.textRange = TextRange.from(0, 0);
     }
```

Comment:
Do we need this previous constructor? Will this end up confusing users/making things error prone?

Reply:
There are places where the `TextRange` information is not available when constructing `BLangDiagnosticLocation`. But we can pass `0`s as the values to TextRange for those cases and remove the old constructor. Should we proceed with that?

Reply:
@dulajdilshan WDYT? Is `BLangDiagnosticLocation` only used in ballerina-lang?

Reply:
Yes, there are places we use `BlangDiagnosticLocation` outside of ballerina-lang. If we remove the previous one, the extensions also needed to be changed suddenly.  I think it would be better if we can have both constructors for now. Let's deprecate the previous one.

Reply:
Updated